src/engine.h

```diff
@@ -53,7 +53,7 @@ class Engine {
     // blocking call to wait for search to finish
     void wait_for_search_finished();
     // set a new position, moves are in UCI format
-    void set_position(const std::string& fen, const std::vector<std::string>& moves);
+    Square set_position(const std::string& fen, const std::vector<std::string>& moves);
```

Comment:
Perhaps a comment can be added here to specify what `set_position()` returns, as it's not immediately clear.